orion-server/src/main/java/com/pinterest/orion/core/automation/operator/kafka/BrokersetTopicOperator.java

```diff
@@ -166,6 +166,14 @@ public void operate(KafkaCluster cluster) throws Exception {
 
       if (idealPartitionCount > actualPartitionCount) {
         // expansion
+        // reject expasion if topic has compact config
+        if (topicAssignment.getConfig() != null &&
+                topicAssignment.getConfig().containsKey("cleanup.policy") &&
+                topicAssignment.getConfig().get("cleanup.policy").equals("compact")) {
+          logger.warning(
+                  "Compact topic " + topicName + " partition count can't be increased - rejecting expansion");
```

Comment:
Would be good to add actual and ideal partition counts in the log message.
Also, `Compacted topic` is the term that is used when compaction is enabled for a topic.